doc/qos/new-headroom-calculation.md

```diff
@@ -0,0 +1,168 @@
+# Requirement
+## Current Solution
+
+- When system start, it reads the pg_profile_lookup.ini and generate an internal lookup table indexed by speed and cable length, and containing size, xon, xoff and threshold.
+- When a port's cable length updated, it records the cable length of the port. But it doesn't update relavent tables accordingly.

```

Comment:
are you saying there is a bug here? "it doesn't update the relevant tables?"

Reply:
IMO, it's a bug. The BUFFER_PG table should be updated accordingly to the port's cable length and speed after they have been updated.